test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -134,37 +133,36 @@ class runtime_monitor : public component {
     {
         configuration *sub_config;
         std::string statistic_list;
-        /* Parse the configuration for the runtime monitor. */
-        _ops = _config->get_int(RATE_PER_SECOND);
 
-        /* Load known statistics. */
-        sub_config = _config->get_subconfig(STAT_CACHE_SIZE);
-        _stats.push_back(new cache_limit_statistic(sub_config));
-        delete sub_config;
+        /* Load the general component things. */
         component::load();
+
+        if (_enabled) {
+            _session = connection_manager::instance().create_session();
+
+            /* Open our statistic cursor. */
+            _session->open_cursor(_session, STATISTICS_URI, nullptr, nullptr, &_cursor);
+
+            /* Load known statistics. */
+            sub_config = _config->get_subconfig(STAT_CACHE_SIZE);
+            _stats.push_back(new cache_limit_statistic(sub_config));
+            delete sub_config;
+        }
     }
 
     void
-    run()
+    do_work()
     {
-        WT_SESSION *session = connection_manager::instance().create_session();
-        WT_CURSOR *cursor = nullptr;
-
-        /* Open a statistics cursor. */
-        testutil_check(session->open_cursor(session, STATISTICS_URI, nullptr, nullptr, &cursor));
-
-        while (_running) {
-            /* Sleep so that we do x operations per second. To be replaced by throttles. */
-            std::this_thread::sleep_for(std::chrono::milliseconds(1000 / _ops));
-            for (const auto &it : _stats) {
-                if (it->is_enabled())
-                    it->check(cursor);
-            }
+        for (const auto &it : _stats) {
+            if (it->is_enabled())
+                it->check(_cursor);
         }
     }
 
     private:
-    int64_t _ops;
+    WT_CURSOR *_cursor = nullptr;
```

Comment:
Sometimes we init class members as nullptr here, sometimes in the constructor init list. See test.h